src/cpufreq.c

```diff
@@ -82,9 +154,64 @@ static int cpufreq_read(void) {
     /* convert kHz to Hz */
     v.gauge *= 1000.0;
 
-    cpufreq_submit(i, v);
-  }
+    cpufreq_submit(i, "cpufreq", NULL, v);
+
+    /* Read total transitions for cpu frequency */
+    if (report_total_trans) {
+      snprintf(filename, sizeof(filename),
+               "/sys/devices/system/cpu/cpu%d/cpufreq/stats/total_trans", i);
+      fh = fopen(filename, "r");
+      if (fh == NULL)
+        continue;
+      while (fgets(buffer, sizeof(buffer), fh) != NULL) {
+        if (!sscanf(buffer, "%lli", &t)) {
+          fclose(fh);
+          return 0;
+        }
+        snprintf(buffer, sizeof(buffer), "%lli", t - t_data[i].transitions);
+        t_data[i].transitions = t;
+      }
+      if (parse_value(buffer, &v, DS_TYPE_GAUGE) != 0) {
+        WARNING("cpufreq plugin: Reading \"%s\" failed.", filename);
+        fclose(fh);
+        continue;
+      }
+      fclose(fh);
+
+      cpufreq_submit(i, "transitions", NULL, v);
+    }
 
+    /* Determine time in state for cpu during previous interval
+     * Reported in 10mS as unit.
+     */
```

Comment:
If unit is 10ms, maybe we have to multiply value to get value in milliseconds?
Similar is done with frequency (kHz converted to Hz).
This is preferred to use.

Reply:
fixed in next revision of patch